test/subscription_map.hpp

```diff
@@ -472,16 +480,24 @@ class multiple_subscription_map
 
         auto& subscription_set = h_iter->second.value;
 
-        auto new_pair = std::make_pair(key, std::forward<V>(value));
-        auto insert_result = subscription_set.insert(new_pair);
-        if (insert_result.second) {
+        // I have to do a find first, because insert_or_assign is available in C++17 only
+        // https://www.fluentcpp.com/2018/12/11/overview-of-std-map-insertion-emplacement-methods-in-cpp17/
```

Comment:
I suggest putting in an ifdef for if we're compiling with C++17 because using the built-in functionality of the class's insert_or_assign function would probably be more performant than what client code is required to do if compiling in C++14 mode.